### ISSUE TITLE
Symbolic Whitespace

### ISSUE THREAD

**Issue by RobertBColton** [TomasMikula/RichTextFX#174]
I am not posting this as an RFE but rather as an inquiry to see if such support is already possible with RTFX to implement easily. In a lot of code editors such as Visual Studio, Eclipse, or Notepad++ there exists the ability and option to visualize white space with symbols. Tabs are typically represented as right pointing arrows and spaces as vertical centered dots.

![Symbolic Whitespace in Eclipse](http://mcuoneclipse.files.wordpress.com/2012/02/show-whitespace-12-02-2012.png)

**Comment by TomasMikula** [comment]
Hi, it is not possible to implement easily, but @JFormDesigner implemented it recently in [Markdown Writer FX](https://github.com/JFormDesigner/markdown-writer-fx) via reflection and misuse of paragraph graphic.

![screen shot 2015-09-01 at 1 40 10 pm](https://cloud.githubusercontent.com/assets/433439/9611686/0dcad84a-50af-11e5-9270-35cca2b3934d.png)

**Comment by RobertBColton** [comment]
@TomasMikula Hi Tomas, thank you for your feedback. The way you describe it makes me ambivalent about taking his approach though. I also see #87 which is somewhat related to this and seems as though you yourself would like a better solution for things like this. Would you like to keep this open as an enhancement or close it and let me link this from #87 as a consideration?

**Comment by TomasMikula** [comment]
You are right, I don't want to encourage people to depend on internals and I would like a better solution. I don't think #87 would be a good way to implement this either, because #87 is about having Nodes as legitimate parts of the document, while here the document would still contain only text and the symbolic whitespace is about styling.

**Comment by afester** [comment]
This is an essential feature for every code editor. If not implemented in the core, we should at least provide a sample implementation to show how to do it.

**Comment by JordanMartinez** [comment]
I think it should be implemented in the core.
